Add handedness guide and link from viewer tutorial

jni opened the pull request:
Closes #570

jni committed:
Add handedness guide and link from viewer tutorial

Closes #570

jni commented:
Only thing I'm unhappy about here is that the stick model is too sparse to be clearly winding in one direction or the other:

![CleanShot 2025-05-01 at 00 17 16@2x](https://github.com/user-attachments/assets/45861f38-c311-42a4-bead-94dc68d0fead)

However, since the code can be run, I'm hoping that it's clear enough and folks can experiment with it. But happy to take suggestions about other examples I could use here...

jni committed:
Merge branch 'main' into handed

jni committed:
Clear double reference

jni committed:
Close viewer at end of guide

psobolewskiPhD commented:
@jni how hard would it be to use the 3d model:
https://www.rcsb.org/3d-view/1BNA
It looks a lot better in terms of "seeing" the threads of the screw.
they have obj and str export...

psobolewskiPhD commented on docs/guides/handedness.md:
````diff
@@ -0,0 +1,169 @@
+
+viewer.close()
+```
+
````
```suggestion
(3D-handedness) = 
```
so we can link here direct

psobolewskiPhD commented on docs/guides/handedness.md:
```diff
@@ -0,0 +1,169 @@
+
+In 3D, an additional *depth* axis is *pre-*pended to the other two. In versions
+0.5.6 and earlier, it pointed *away* from the camera, while in versions 0.6.0
+it points *towards* the camera.
```
````suggestion
In 3D, an additional *depth* axis is *pre-*pended to the other two. 
```{important}
In versions 0.5.6 and earlier, it pointed *away* from the camera, while in versions 0.6.0
it points *towards* the camera. This affects how 3D images and models are rendered, see
[3D data, 3D axis orientation, and handedness](3D-handedness).
```
````

psobolewskiPhD commented on docs/guides/handedness.md:
````diff
@@ -0,0 +1,169 @@
+```
+
+If we flip one of the axes, we will get the mirror image of the DNA, which
+will be physically inaccurate:
````
```suggestion
Starting with napari 0.6.0, using the default orientations, the above image will correctly represent
a right-handed helix. However, if we flip one of the axes (as they were in napari prior to 0.6.0), 
we will get the mirror image of the DNA, which will be physically inaccurate:
```

psobolewskiPhD commented on docs/guides/handedness.md:
````diff
@@ -0,0 +1,169 @@
+viewer.close()
+```
+
+The camera controls, accessible by right-clicking the 2D/3D toggle button,
+will display whether your current axis orientations form a [right-handed or a
+left-handed coordinate frame][wiki-right-hand-rule].
+
+![camera controls, with a "right handed reference frame" icon highlighted
+  next to the axis orientation boxes](../_static/images/handedness.png)
````
````suggestion
```{tip}
The camera controls, accessible by right-clicking the 2D/3D toggle button,
will display whether your current axis orientations form a [right-handed or a
left-handed coordinate frame][wiki-right-hand-rule].

![camera controls, with a "right handed reference frame" icon highlighted
  next to the axis orientation boxes](../_static/images/handedness.png)
```
````
I love me some admonitions!

TimMonko commented on docs/guides/handedness.md:
````diff
@@ -0,0 +1,169 @@
+```{code-cell} python
+:tags: [remove-input]
+
+viewer.camera.angles = (92, 0, 0)
````
```suggestion
viewer.camera.angles = (90, 0, 90)
```

For simplicity, changing camera angles to (90,0,90) seems to improve readability quite a lot (for a nice fast merge)

![image](https://github.com/user-attachments/assets/0fa44129-6984-4f0a-8a8a-30a1c1c7a39c)

![image](https://github.com/user-attachments/assets/0aed2562-4561-4334-b137-6a34967d6e80)

TimMonko reviewed (approved):
Overall, great! Agree with @psobolewskiPhD that it's a bit hard to 'see'.

With my suggestion I think this is sufficient.

psobolewskiPhD reviewed (approved):
I made some suggestions to the copy.
I think the first admonition is important, I try to make it clear that 0.6.0 does 3D different.
I like admonitions in general, so I added another :P

jni commented:
Ah nice, obj is probably doable since Vispy has a reader. Gonna try now. Thanks both!

jni commented:
(Just as an aside, I also explored using the points layer for the atoms, but all the stupid hydrogen atoms floating around made it all muddled anyway LOL)

jni commented:
Lookin good...

![CleanShot 2025-05-01 at 12 03 51@2x](https://github.com/user-attachments/assets/50f46556-bc04-487d-9e1d-be02a53dcb0c)

jni committed:
Emphasise 0.5.6- and 0.6.0+ differences with admonition

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

jni committed:
Place camera controls image in admonition

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

jni committed:
Use surface model instead of bond data

This makes the difference between right handed and
left handed helices much more obvious

jni committed:
Update handedness icon image

jni committed:
Fix myst target declaration

jni committed:
fix markdown formatting

jni commented:
Current look:

![CleanShot 2025-05-01 at 12 35 44@2x](https://github.com/user-attachments/assets/9271bec0-8534-41d0-8a53-806e9479fb9f)

I think this should be ready-to-merge! Pending build...

jni commented:
Thanks @psobolewskiPhD for the model suggestion, I didn't know I could export that, it makes the tutorial waaaay better! 😍

jni committed:
Add smooth shading

jni commented:
The little triangles were bugging me (even if they tied in nicely with the triangulation discussion 😂) so I added smooth shading to the rendering. Looks quite a bit nicer this way 😃 

![handedness](https://github.com/user-attachments/assets/4069198b-8675-4dc5-9315-747828835c7e)

psobolewskiPhD commented:
Looks great  😊
